### PR TITLE
feat: Implement Replay Start endpoint

### LINKED ISSUE #11: Implement the Replay Start Endpoint
Per ADR: https://docs.edgexfoundry.org/3.0/design/adr/application/0025-Record-and-Replay/#post_1

Unit test endpoint using Data Manager mock

### PR DESCRIPTION
**Pull request by lenny-intel**
closes #11

**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/app-record-replay/blob/main/.github/CONTRIBUTING.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [x] I have added unit tests for the new feature or bug fix (if not, why?)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?) **Not Yet**
  <link to docs PR>

## Testing Instructions
Run non-secure EdgeX stack
Run `make build`
Run app service with `-cp -d`
Use postman collecting to test OST to `localhost:59712/api/v3/replay`

### CODE REVIEW

**Review by marcpfuller — APPROVED** [review]
LGTM

File: internal/controller/controller.go [hunk 1]
```diff
@@ -39,6 +39,9 @@ const (
 	failedRecordDurationValidate   = "Record request failed validation: Duration must be > 0 when set"
 	failedRecordEventLimitValidate = "Record request failed validation: Event Limit must be > 0 when set"
 	failedRecording                = "Recording failed"
+	failedReplayRateValidate       = "Replay request failed validation: Replay Rate must be less than or greater than 0"
```
[the remark this answers is not in the document]
Comment (lenny-intel) [reply internal/controller/controller.go:42]: I like it!

[the remark this answers is not in the document]
Comment (lenny-intel) [reply internal/controller/controller.go:42]: Actually, validation it just must be > 0

[the remark this answers is not in the document]
Comment (lenny-intel) [reply internal/controller/controller.go:42]: rework

File: internal/controller/controller.go [hunk 2]
```diff
@@ -161,8 +164,33 @@ func (c *httpController) recordingStatus(writer http.ResponseWriter, request *ht
...
-	writer.WriteHeader(http.StatusNotImplemented)
+	startRequest := &dtos.ReplayRequest{}
+
+	if err := json.NewDecoder(request.Body).Decode(startRequest); err != nil {
+		writer.WriteHeader(http.StatusBadRequest)
+		_, _ = writer.Write([]byte(fmt.Sprintf("%s: %v", failedRequestJSON, err)))
+		return
+	}
+
+	if startRequest.ReplayRate == 0 {
```
Comment (jim-wang-intel) [inline internal/controller/controller.go:175]: you meant <= 0? 
```suggestion
	if startRequest.ReplayRate <= 0 {
```

Comment (lenny-intel) [reply internal/controller/controller.go:175]: Good catch

Comment (lenny-intel) [reply internal/controller/controller.go]: fixed

**Review by jim-wang-intel — APPROVED** [review]
LGTM

**Pull request merged**